### ISSUE TITLE
Specify relation/constraint between interactions with math

### ISSUE THREAD

**Issue by miguelrk** [w3c/wot-thing-description#1253]
**TLDR: The TD could add a mechanism of specifying constraints and/or relations between interactions with math.**

The goal of a Thing Description is describing the interaction affordances of things. Currently, one can only describe them independently and cannot describe relations between them. In reality though, there are many cases where one could indeed define a hard constraint (or heuristic), to allow coupling the description of interaction affordances.

**Lets consider an example of a fan:**

We know that if the fan has `status === 'off'`, the speed _should_ in an ideal world be 0 (_should_ since this is obviously dependent on external factors to the system, e.g. there might be wind causing it to turn). 

The TD _could_ have a way of describing Data Schemas not only in isolation, but also in relation to other Data Schemas within the TD (or maybe even to other TDs). 

This could be done by adopting a query/mathematical notation like [MathJSON](https://cortexjs.io/math-json/) or maybe even JS expressions (though not language agnostic anymore, and might imply security risk).
```json
{
  "title": "Fan",
  "properties": {
    "status": { "type": "string", "enum": ["on", "off"] },
    "speed": { "type": "number", "constraints": ["properties.status === 'off' ? 0 : properties.speed"] }
  }
}
```

**Lets consider another example of a Pendulum:**

Let's say we only have one sensor for `velocity v` and we know the static `mass` value (12kg), but we also want to expose `momentum p = mv` as a property affordance of the thing. Exposing all 3 as properties, would make them accessible which might be desired. We could write something like:
```json
{
  "title": "Cart",
  "properties": {
   "mass": { "type": "number", "const": 12, "unit": "kg", "readOnly": true },
    "velocity": { "type": "number", "enum": ["on", "off"] },
    "momentum": { "type": "number", "readOnly": true, "constraints": ["properties.velocity*properties.mass"] }
  }
}
```

For this, I think there is nothing for succinct than plain mathematical expressions/equations.

One could argue that a proper Thing equipped with working sensors would always provide the `status` and `speed` value in real time. The more compelling use case for this additional 'relations' description might be for virtual/simulated Things, and for UI generation, where this extra information could be used.

NOTE: Being able to specify a `const` in a Data Schema (see `mass` in the second example above) would also be requried for these type of cases, although properties are usually though of as dynamic state, and not static. But where could one else add this static information inside of a TD in order to also make it available
